lib/streamlit/runtime/uploaded_file_manager.py

```diff
@@ -61,9 +69,17 @@ class UploadedFileManager(CacheStatsProvider, Protocol):
     """# TODO(kajarenc): Docstrings for this protocol + its methods."""
 
     @abstractmethod
-    def get_files(self, session_id: str, file_urls: List[str]) -> List[UploadedFileRec]:
+    def get_files(
+        self, session_id: str, file_ids: Sequence[str]
+    ) -> List[UploadedFileRec]:
         raise NotImplementedError
 
     @abstractmethod
     def remove_session_files(self, session_id: str) -> None:
         raise NotImplementedError
+
+    @abstractmethod
+    def get_upload_urls(
+        self, session_id: str, file_names: Sequence[str]
+    ) -> List[UploadFileUrlInfo]:
+        raise NotImplementedError
```

Comment:
nit: I wonder if we want to leave the `@abstractmethod` decorator off this one since some implementations of `UploadedFileManager` may choose to leave this unimplemented.

We'd probably still want to `raise NotImplementedError` as the default implementation, though, so that things do throw an error if an intentionally unimplemented method is somehow called